src/internal.c

```diff
@@ -9405,18 +9379,25 @@ static void DoCertFatalAlert(WOLFSSL* ssl, int ret)
  * store->error_depth member to determine index (0=peer, >1 intermediates)
  */
 
-static int DoVerifyCallback(WOLFSSL* ssl, int ret, ProcPeerCertArgs* args)
+int DoVerifyCallback(WOLFSSL_CERT_MANAGER* cm, WOLFSSL* ssl, int ret,
+                                                        ProcPeerCertArgs* args)
 {
     int verify_ok = 0, use_cb = 0;
-
+#ifdef WOLFSSL_SMALL_STACK
+    void *heap = (ssl != NULL) ? ssl->heap : cm->heap;
+#endif
     /* Determine if verify was okay */
     if (ret == 0) {
         verify_ok = 1;
     }
 
     /* Determine if verify callback should be used */
     if (ret != 0) {
-        if (!ssl->options.verifyNone) {
+        if (((ssl != NULL) && (!ssl->options.verifyNone))
+#ifndef NO_WOLFSSL_CM_VERIFY
+            || ((cm != NULL) && (cm->verifyCallback != NULL))
```

Comment:
The `use_cb` here is meant for the ssl only. I recommend reverting this one and cleaning up the logic at 9452.